modin/core/execution/ray/generic/io/io.py

```diff
@@ -118,9 +118,23 @@ def to_csv(cls, qc, **kwargs):
         if not cls._to_csv_check_support(kwargs):
             return BaseIO.to_csv(qc, **kwargs)
 
-        # The partition id will be added to the queue, for which the moment
-        # of writing to the file has come
-        queue = Queue(maxsize=1)
+        @ray.remote
+        class SignalActor:  # pragma: no cover
+            def __init__(self, event_count):
+                self.events = [asyncio.Event() for _ in range(event_count)]
+
+            def send(self, event_idx):
+                self.events[event_idx].set()
+
+            async def wait(self, event_idx, should_wait=True):
+                if should_wait:
+                    await self.events[event_idx].wait()
```

Comment:
```suggestion
            async def wait(self, event_idx):
                await self.events[event_idx].wait()
```
I don't see any point of passing this argument - if it should be `False` one can just _not_ call the method.
Plus I don't see it used anywhere, so let's keep it a little cleaner.